src/compiler/checker.ts

```diff
@@ -17925,7 +17925,7 @@ namespace ts {
                         }
                     }
                 }
-                else if (isGenericMappedType(target) && !target.declaration.nameType) {
+                else if (isGenericMappedType(target) && (!target.declaration.nameType || relation === comparableRelation)) {

```

Comment:
Without this change and the bit below, you couldn't cast between an `Omit` of something and that something. That mapped types with `nameType` fields currently have no generic relations is a place we should probably improve upon anyway, however this is the minimal change required to get that comparable relation to work as expected.